server/src/main/java/org/tctalent/server/service/db/UserService.java

```diff
@@ -87,7 +87,7 @@ public interface UserService {
      * Checks the validity of the email verification token and verifies the user's email.
      * Throws an error if the email verification token is invalid or expired, or if sending the completion verification email fails.
      */
-    void verifyEmail(VerifyEmailRequest request);
+    User verifyEmail(VerifyEmailRequest request);
```

Comment:
I don't think that this is the best way to get the User because you don't get any user if the token as expired. See comment above.